bad_data/README.md

```diff
@@ -22,3 +22,5 @@ These are files used for reproducing various bugs that have been reported.
 
 * PARQUET-1481.parquet: tests a case where a schema Thrift value has been
   corrupted
+
+* ARROW-GH-43605.parquet: In Go, file reader goroutine crashed.
```

Comment:
Can you give me some help to know which part of the file is corrupted? I am not expert enough to know why the file is corrupt. I only know it is generated with a specific version of polars, as described in this PR: https://github.com/apache/arrow/pull/43607#issuecomment-2275701780
Latest version of polars does not cause the issue anymore.

Reply:
Wait a minute, I'll test this on my local pc

Reply:
Emm I don't know, arrow-c++ can read this file, emm...

Reply:
If arrow-cpp manages to open it is nice, arrow-go crashed with this before related PR on arrow. Should I move this into valid data folder then?

Reply:
The criterion should be: is a valid Parquet file or not?

Reply:
If it's a valid Parquet file and the issue was a bug in Arrow Go preventing to read it, then it should go in `data`. If it's an invalid Parquet file and the issue was that Arrow Go crashed instead of reporting a regular error, then it should go in `bad_data`.

Reply:
I'll check this later

Reply:
See: https://github.com/apache/parquet-testing/pull/57#issuecomment-2307509744

Dictionary with RLE encoding bit-width is 0.